### ISSUE TITLE
Emit error when file-level `# pyright` comment does not appear on its own line

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4982]
**Is your feature request related to a problem? Please describe.**
`# pyright: reportGeneralTypeIssues=false` and `# pyright: ignore[reportGeneralTypeIssues] `can feel similar to one less familiar with pyright comment exact meaning but the former is really heavy and removes rule entire file. That is expected behavior and I understand that for some files it can be useful to disable some rules. For repository I work on I'd like to mostly block file rule overrides and encourage instead people ignore affected lines as needed.

The one awkward rule is reportImportCycles. That rule is nice but some older files in codebase I work on need it and it can't be ignored in a line level.

**Describe the solution you'd like**
A configuration setting to disallow/ignore file level overrides for every rule except reportImportCycles. I'm also happy with having it cover reportImportCycles and I can do a separate pyright pass just for that rule/different linter for import cycles.

**Additional context**
Minor request. Just something I noticed and was confused by today why basic type error in a line was being missed and then realized someone put `# pyright: reportGeneralTypeIssues=false` on another line intending to ignore one error. This can also be covered by remembering to check it in pr reviews but in a repo shared by many it'd easier to make it harder to happen to ignore 10+ errors at once.

**Comment by erictraut** [comment]
I don't think this feature makes sense. These comments are supported for good reason. I don't see a need to disable them. This is something we could revisit in the future if we get signal from other pyright users, but for now I'm going to close this.

**Comment by hmc-cs-mdrissi** [comment]
I can say I see like ~30ish usages of `# pyright: reportGeneralTypeIssues ` being used in 1 line when pyright ignore looks likely to be intent from different contributors.

An alternative proposal can # pyright: reportFoo be restricted to top of file? If someone does,

```python
# pyright: reportGeneralTypeIssues=False

def foo(x):
  ...

class Bar:
  ...
```

then they probably intended file level disabling. But if someone writes,

```python
def foo(x):
  z: str = 3 # pyright: reportGeneralTypeIssues=False

class Bar:
  ...
```

then I'd be surprised if they really intended file disable and my guess is they just copied from another file and misunderstood effect.

**Comment by erictraut** [comment]
The `# pyright: reportXXX=false` comment form should always appear on its own line, not after a Python statement. It's not important that it appears at the top of a file, but it should be on its own line. I think it's reasonable to emit an error when it occurs at the end of a line. Does that sound like a good solution to you?

**Comment by hmc-cs-mdrissi** [comment]
I'd be happy with that. That would be enough to cover most misusages I see in my codebase.

**Comment by hmc-cs-mdrissi** [comment]
One optional extension, I occasionally see usages that look like,

```python
class Foo:
  def method1(self):
     # pyright: reportXXX=false
     ...

  def method2(self):
    # pyright: reportXXX=false
    ...
```

which looks like the intent is block/function scoped ignore. I also see in sourcegraph sometimes this,


```python
import ..

class Foo:
  # pyright: reportXXX=false
  ...
```

which looks class scoped ignore. Given that ignore affects entire file can it also emit an error if comment is not defined in global/module scope?

Regardless restricting it to it's own line catches most of misusages I see and happy if you don't want to add a scope rule.

**Comment by erictraut** [comment]
Comments don't have scopes. Scopes are defined by a parse tree, and comments are not parsed. Comments are associated with tokens, not with parse nodes.

I could enforce that comments that control file-scoped rules always start at the beginning of line (at column 1).

**Comment by hmc-cs-mdrissi** [comment]
Column 1/not indented sounds like a good way.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.305, which I just published. It will also be included in a future release of pylance.
